src/dialogue.h

```diff
@@ -81,98 +81,103 @@ struct talk_topic {
     std::string reason;
 };
 
+struct effect_fun_t {
+    private:
+        std::function<void( const dialogue &d )> function;
+
+    public:
+        effect_fun_t() = default;
+        effect_fun_t( talkfunction_ptr effect );
+        effect_fun_t( const std::function<void( npc & )> effect );
+        void set_companion_mission( const std::string &role_id );
+        void set_u_add_permanent_effect( const std::string &new_effect );
+        void set_u_add_effect( const std::string &new_effect, const time_duration &duration );
+        void set_npc_add_permanent_effect( const std::string &new_effect );
+        void set_npc_add_effect( const std::string &new_effect, const time_duration &duration );
+        void set_u_add_trait( const std::string &new_trait );
+        void set_npc_add_trait( const std::string &new_trait );
+        void set_u_buy_item( const std::string &new_trait, int cost, int count,
+                             const std::string &container_name );
+        void set_u_spend_cash( int amount );
+        void set_npc_change_faction( const std::string &faction_name );
+        void set_change_faction_rep( int amount );
+        void operator()( const dialogue &d ) const {
+            if( !function ) {
+                return;
+            }
+            return function( d );
+        }
+};
+
 /**
- * This defines possible responses from the player character.
+ * Defines what happens when the trial succeeds or fails. If trial is
+ * TALK_TRIAL_NONE it always succeeds.
  */
-struct talk_response {
+struct effect_t {
```

Comment:
This in the global scope can be a bit confusing as there are already completely unrelated classes `effect_type` and `effect`.

Reply:
Changed to talk_effect_t and talk_effect_fun_t, thanks for pointing that out.